regex-syntax/src/lib.rs

```diff
@@ -1589,8 +1589,7 @@ impl fmt::Display for ErrorKind {
             RepeaterExpectsExpr =>
                 write!(f, "Missing expression for repetition operator."),
             RepeaterUnexpectedExpr(ref e) =>
```

Comment:
I didn't remove the unused variable `e` because I assumed it can be put back into the error message again once the issues with the AST pretty printing are solved. But, if it makes sense, I can modify the code further to convert `RepeaterUnexpectedExpr(Expr)` to `RepeaterUnexpectedExpr` and get rid of the unused variable.

Reply:
@andrea-prearo Right, but this produces a compile warning, which causes the build to fail. You definitely don't want to change the error type itself, but using `RepeaterUnexpectedExpr(_)` is the right thing to do here.

There are other errors that try to print sub-expressions. Could you fix those too? e.g., `InvalidClassEscape` and maybe more.

Reply:
@BurntSushi I just noticed that warnings are treated as errors by looking at the CI log. I'll fix that by using `RepeaterUnexpectedExpr(_)` as you suggested. I'll also work on fixing other sub-expressions.